### PR TITLE
Fix issues in inputs

### PR DESCRIPTION
**Pull request by totop716**
- Create `NumericalInput` component for the input that only allows decimals
- Replace the inputs with `NumericalInput` for input for swap, adding/removing liquidity, staking/unstaking QUICK, staking in syrup, staking/unstaking in farms.
- Remove and clear Recipient input when expert mode is turned off.
- Add placeholder to Recipient input.
- Add new farms, syrups, tokens and token icons.
- Show error on swap button when the swap is not valid

### CODE REVIEW

File: src/components/FarmDualCard/FarmDualCardDetails.tsx [hunk 1]
```diff
@@ -403,11 +393,12 @@ const FarmDualCardDetails: React.FC<{
...
             </Box>
             <Box className={classes.inputVal} mb={2} mt={2} p={2}>
-              <input
+              <NumericalInput
                 placeholder='0.00'
                 value={stakeAmount}
-                onChange={(evt: any) => {
-                  setStakeAmount(evt.target.value);
+                fontSize={16}
+                onUserInput={(value) => {
```
Comment (FriskyHamTitz) [inline src/components/FarmDualCard/FarmDualCardDetails.tsx:400]: is onUserInput the equivalent to onChange?

File: src/components/SettingsModal/SettingsModal.tsx [hunk 2]
```diff
@@ -275,14 +282,17 @@ const SettingsModal: React.FC<SettingsModalProps> = ({ open, onClose }) => {
...
                 value={slippageInput}
+                fontSize={14}
+                fontWeight={500}
+                align='right'
+                color='rgba(212, 229, 255, 0.8)'
                 onBlur={() => {
                   parseCustomSlippage((userSlippageTolerance / 100).toFixed(2));
                 }}
-                onChange={(e: any) => parseCustomSlippage(e.target.value)}
+                onUserInput={(value) => parseCustomSlippage(value)}
```
Comment (FriskyHamTitz) [inline src/components/SettingsModal/SettingsModal.tsx:295]: Is on user input a simplification so we don have to go through `e.Target`

can `value` be typed?

File: src/components/StakeQuickModal/StakeQuickModal.tsx [hunk 3]
```diff
@@ -146,19 +136,17 @@ const StakeQuickModal: React.FC<StakeQuickModalProps> = ({ open, onClose }) => {
...
           </Box>
           <Box mt={2} display='flex' alignItems='center'>
-            <input
+            <NumericalInput
               placeholder='0'
-              className={classes.input}
               value={typedValue}
-              onChange={(evt: any) => {
+              fontSize={28}
+              onUserInput={(value) => {
```
Comment (FriskyHamTitz) [inline src/components/StakeQuickModal/StakeQuickModal.tsx:143]: Now that were using a numerical input can't value be typed already?

File: src/components/Swap/Swap.tsx [hunk 4]
```diff
@@ -618,37 +619,31 @@ const Swap: React.FC<{
...
-            <ArrowDown size='16' color='white' />
+            {recipient !== null ? (
+              <ArrowDown size='16' color='white' />
+            ) : (
+              <Box />
+            )}
             <Button
               id='remove-recipient-button'
-              onClick={() => onChangeRecipient(null)}
+              onClick={() => onChangeRecipient(recipient !== null ? null : '')}
```
Comment (FriskyHamTitz) [inline src/components/Swap/Swap.tsx:632]: I'm pretty sure we still need to clear the recipient when removing.
e.g if there is a recipient, and then they turn expert mode off will it still send to that recipient?

**Review by FriskyHamTitz — CHANGES_REQUESTED** [review]

File: src/components/FarmDualCard/FarmDualCardDetails.tsx (hunk 1 above)
Comment (totop716) [reply src/components/FarmDualCard/FarmDualCardDetails.tsx:400]: onUserInput is equivalent to onChange but it returns the typed string as parameter

File: src/components/SettingsModal/SettingsModal.tsx (hunk 2 above)
Comment (totop716) [reply src/components/SettingsModal/SettingsModal.tsx:295]: Yes, the value is the string typed in the input

File: src/components/Swap/Swap.tsx (hunk 4 above)
Comment (totop716) [reply src/components/Swap/Swap.tsx:632]: It removes the recipient when toggling off the expert mode

File: src/components/StakeQuickModal/StakeQuickModal.tsx (hunk 3 above)
Comment (totop716) [reply src/components/StakeQuickModal/StakeQuickModal.tsx:143]: The value is returning the typed string in the input

**Pull request merged**